ksiren/src/main/java/com/brightspace/ksiren/JsonUtils.kt

```diff
@@ -16,70 +18,160 @@ package com.brightspace.ksiren
  * limitations under the License.
  */
 internal object JsonUtils {
-	fun toJson(entity: Entity) = listOf(
-			"class" to entity.classes.toJson(),
-			"rel" to entity.rel.toJson(),
-			"properties" to entity.enhancedProperties.toJson(),
-			"entities" to entity.entities.toJson(),
-			"actions" to entity.actions.values.toJson(),
-			"links" to entity.links.toJson()
-		)
-		.toJsonObject()
-
-	fun toJson(link: Link) = listOf(
-			"rel" to link.rels.toJson(),
-			"class" to link.classes.toJson(),
-			"title" to link.title.toJson(),
-			"type" to link.type.toJson(),
-			"href" to link.href.toJson()
-		).toJsonObject()
-
-	fun toJson(action: Action) = listOf(
-			"name" to action.name.toJson(),
-			"title" to action.title.toJson(),
-			"class" to action.classes.toJson(),
-			"method" to action.method.toJson(),
-			"href" to action.href.toJson(),
-			"type" to action.type.value.toJson(),
-			"fields" to action.fields.toJson()
-		).toJsonObject()
-
-	fun toJson(field: Field) = listOf(
-			"name" to field.name.toJson(),
-			"class" to field.classes.toJson(),
-			"type" to field.type.toJson(),
-			"value" to field.value.toJson()
-		).toJsonObject()
-
-	fun toJson(propertyValue: PropertyValue): String? = when(propertyValue) {
-		is StringValue -> propertyValue.stringValue.toJson()
-		is BooleanValue -> propertyValue.booleanValue.toString()
-		is ArrayValue -> propertyValue.arrayElements.toJson()
-		is ObjectValue -> propertyValue.objectProperties.toJson()
+	fun toJson(entity: Entity, writer: KSirenJsonWriter): String {
+		writer.use {
+			writeEntity(entity, writer)
+		}
+
+		return writer.getSerializedString()
+	}
+
+	fun toJson(action: Action, writer: KSirenJsonWriter): String {
+		writer.use {
+			writeAction(action, writer)
+		}
+
+		return writer.getSerializedString()
+	}
```

Comment:
nit: If `getSerializedString()` works before close.

```suggestion
	fun toJson(action: Action, writer: KSirenJsonWriter) = writer.use {
		writeAction(action, writer)
		writer.getSerializedString()
	}
```